.circleci/config.yml

```diff
@@ -0,0 +1,290 @@
+      - attach_workspace:
+          at: ReactAndroid/src/androidTest/assets/
+      - run:
+          name: Check for JavaScript Bundle
+          command: |
+            if [[ ! -e ReactAndroid/src/androidTest/assets/AndroidTestBundle.js ]]; then
+              echo "JavaScript bundle missing, verify build-js-bundle step"; exit 1;
+            else
+              echo "JavaScript bundle found.";
+            fi
+# CircleCI does not support interpolating env variables in the environment: step above.
+# https://circleci.com/docs/2.0/env-vars/#interpolating-environment-variables-to-set-other-environment-variables
+      - run:
+          name: Configure PATH
+          command: |
+            echo 'export PATH=$ANDROID_NDK:$PATH' >> $BASH_ENV
+            echo 'export PATH=~/react-native/gradle-2.9/bin:~/buck/bin:$PATH' >> $BASH_ENV
+            source $BASH_ENV
+# Configure dependencies
+      - run:
+          name: Install Build Dependencies
+          command: |
+            sudo apt-get update -y
+            sudo apt-get install ant autoconf automake g++ gcc libqt5widgets5 lib32z1 lib32stdc++6 libtool make maven python-dev python3-dev qml-module-qtquick-controls qtdeclarative5-dev -y
+      - run:
+          name: Configure Android SDK and Android Virtual Device
+          command: source scripts/circle-ci-android-setup.sh && getAndroidSDK
+# Starting emulator in advance as it takes some time to boot.
+      - run:
+          name: Launch Android Virtual Device
+          command: emulator64-arm -avd testAVD -no-audio -no-window -no-boot-anim -gpu off
+          background: true
+# Continue configuring dependencies while AVD boots.
+      - restore-cache: *restore-cache-ndk
+      - run:
+          name: Install Android NDK
+          command: |
+            if [[ ! -e ~/ndk ]]; then
+              mkdir ~/ndk
+              cd ~/ndk
+              curl -o ndk.zip --silent https://dl.google.com/android/repository/android-ndk-r10e-linux-x86.zip
+              curl -o ndk_64.zip --silent https://dl.google.com/android/repository/android-ndk-r10e-linux-x86_64.zip
+              unzip -o ndk.zip
+              unzip -o ndk_64.zip
+              rm ndk.zip
+              rm ndk_64.zip
+            fi
+      - save-cache: *save-cache-ndk
+      - restore-cache: *restore-cache-buck
+      - run:
+          name: Install Buck
+          command: |
+            if [[ ! -e ~/buck ]]; then
+              git clone https://github.com/facebook/buck.git ~/buck
+              cd ~/buck
+              git checkout v2017.09.04.02
+            fi
+            cd ~/buck && ant
+            buck --version
+      - save-cache: *save-cache-buck
+      - restore-cache: *restore-cache-watchman
+      - run:
+          name: Install Watchman
+          command: |
+            if [[ ! -e ~/watchman ]]; then
+              mkdir ~/watchman
+              git clone https://github.com/facebook/watchman.git ~/watchman
```

Comment:
`--branch v4.9.0 --depth=1` might speed things up (and similar for the Buck repo clone)